includes/class-wc-gateway-ppec-plugin.php

```diff
@@ -492,6 +494,28 @@ public static function needs_shipping() {
 		return apply_filters( 'woocommerce_cart_needs_shipping', $needs_shipping );
 	}
 
+	/**
+	 * Displays notice to upgrade to PayPal Payments.
+	 *
+	 * @param string $plugin_file Path to the plugin file relative to the plugins directory.
+	 * @param array $plugin_data An array of plugin data.
+	 * @param string $status Status filter currently applied to the plugin list.
+	 */
+	public function ppec_upgrade_notice( $plugin_file, $plugin_data, $status ) {
+		// Return if not on installed plugins page.
+		global $pagenow;
+		if ( 'plugins.php' !== $pagenow ) {
+			return;
+		}
+
+		// Load styles & scripts required for the notice.
+		wp_enqueue_style( 'ppec-upgrade-notice', plugin_dir_url( __DIR__ ) . '/assets/css/admin/ppec-upgrade-notice.css', array(), WC_GATEWAY_PPEC_VERSION );
+		wp_enqueue_script( 'ppec-upgrade-notice-js', plugin_dir_url( __DIR__ ) . '/assets/js/admin/ppec-upgrade-notice.js', array(), WC_GATEWAY_PPEC_VERSION, false );
+
+		// Load notice template.
+		include_once $this->plugin_path . 'templates/paypal-payments-upgrade-notice.php';
```

Comment:
Nice overall approach with this function. I like it 💯.

Reply:
👍 I wanted to keep everything tied to less number of files / functions just in case we want to remove it later.